error/s2n_errno.c

```diff
@@ -255,6 +255,7 @@ static const char *no_such_error = "Internal s2n error";
     ERR_ENTRY(S2N_ERR_UNSUPPORTED_EXTENSION, "Illegal use of a known, supported extension") \
     ERR_ENTRY(S2N_ERR_MISSING_EXTENSION, "Mandatory extension not received") \
     ERR_ENTRY(S2N_ERR_DUPLICATE_EXTENSION, "Extension block contains two or more extensions of the same type") \
+    ERR_ENTRY(S2N_ERR_DEPRECATED_SECURITY_POLICY, "Deprecated security policy. Please choose a modern security policy.") \
```

Comment:
Nit: Is "modern" the right word? You could choose an EXTREMELY old policy if you want, it just can't use a feature that no longer exists. Maybe  "Security policy no longer supported. Please choose a different security policy." I also wonder if we need to update the security policy part of the usage guide to note that this is possible :thinking:

Reply:
Updated message to "different policy" instead of "more modern policy".